client/src/App.js

```diff
@@ -1,13 +1,14 @@
 import React from "react";
-import { HomePage, CoursesPage, Programmes, Students } from "./pages";
+import { HomePage, CoursesPage, ProgrammesPage, Students } from "./pages";
 import { Route, Switch } from "react-router-dom";
 
 const App = () => {
     return (
         <div>
             <Switch>
                 <Route exact path="/" component={HomePage} />
-                <Route path="/programmes/" component={Programmes} />
+                <Route exact path="/programmes/" component={ProgrammesPage} />
+                <Route path="/programmes/:id" component={ProgrammesPage} />
```

Comment:
Both are required (I think) because the ProgrammesPage is the overall page that contains both segments. It needs:
 - the exact path so that it can render an empty programmes page if there are no programmes
 - the path with the id parameter to fetch each programme